### ISSUE TITLE
require-data-selectors rule should not apply to cy.get('@alias')

### ISSUE THREAD

**Issue by judemorrissey** [cypress-io/eslint-plugin-cypress#44]
Hello,

I was in the midst of updating Cypress related packages, and updated this package to the latest 2.9.0 release. I also realized there's the **require-data-selectors** rule which I recently enabled as a warning. However, now I have a whole bunch of warnings for places where I use aliases, e.g.
```javascript
cy.get('[data-test=a]').as('a'); // this is fine
cy.get('@a'); // this is a warning
```

Minor issue as this is a non-essential rule to us, but I thought for posterity it is worth mentioning.

**Comment by SeanPercy** [comment]
I can second that. In my `.eslintrc` file the rule `cypress/require-data-selectors` is set to `1`.
I have a list of selectors that are defined like this:


```
// selectors.js

const addPrefix = testId => `[data-cy=${testId}]`;
export const ASSESSMENT_SUBMIT = addPrefix('assessment-submit');
```

They are imported by my test files and used as followed: 

```
// test.js
cy.get(ASSESSMENT_SUBMIT).click();
```
Running eslint outputs the following:
```
warning  use data-* attribute selectors instead of classes or tag names  cypress/require-data-selectors
```

This is also the case, when they are defined in the same place.

```
const ASSESSMENT_SUBMIT = '[data-cy=assessment-submit]';
cy.get(ASSESSMENT_SUBMIT).click();

```

**Comment by morficus** [comment]
+1 to this. 
I have a similar setup to what @SeanPercy explained where my selectors are external to the spec file and being imported as variables.
